### PR TITLE
Example readme and some light refactoring

### PR DESCRIPTION
**Pull request by mgroeber9110**
I have tried to take a first stab at writing a brief README for main.exe ( #518).

While trying to untangle the ``--instruct`` and ``--interactive`` behaviour, I noticed a small inconsistency in the naming of internal parameters: There is ``params.interactive_start``, while the command line parameter is called ``--interactive_first``, and there is also an ``--interactive_start`` that, confusingly, does the same as ``--interactive``.

So, I added a small refactoring change to rename the structure member for consistency.

Feedback is very welcome. If this looks reasonable, I can also try the same at least for *perplexity* and *embeddings*.

### CODE REVIEW

**Comment by DannyDaemonic** [comment]
This is unfortunate. I was working on this myself. Mine is at #1139.

This is actually pretty decent. I noticed some spelling errors.
seemlessly -> seamlessly
randon -> random
somehwat -> somewhat
candiates -> candidates

**Comment by Green-Sky** [comment]
but does this actually fist interactive-first? i realized it was broken recently

**Comment by mgroeber9110** [comment]
> but does this actually fist interactive-first? i realized it was broken recently

It seems to work for me, at least to the extent where it stops after the first prompt, even if it does not end in a reverse prompt. Is there anything specific where it does not work?

**Comment by mgroeber9110** [comment]
I have merged a few details to the new README that I think might be worth keeping (as it took me a while to figure them out). If this gets too detailed, I am fine with taking them out again.

File: examples/main/README.md [hunk 1]
```diff
@@ -1,65 +1,191 @@
...
+By understanding and using these performance tuning settings, you can optimize the LLaMA model's behavior to achieve the best performance for your specific needs.
+
+## Additional Options
+
+These options provide extra functionality and customization when running the LLaMA models:
+
+-   `-h, --help`: Display a help message showing all available options and their default values. This is particularly useful for checking the latest options and default values, as they can change frequently, and the information in this document may become outdated.
+-   `--verbose-prompt`: Print the prompt before generating text.
+-   `--mtest`: Test the model's functionality by running a series of tests to ensure it's working properly.
+-   `--lora FNAME`: Apply a LoRA (Layer-wise Relevance Approximation) adapter to the model (implies --no-mmap). This allows you to adapt the pretrained model to specific tasks or domains.
```
Comment (slaren) [inline examples/main/README.md]:
```suggestion
-   `--lora FNAME`: Apply a LoRA (Low-Rank Adaptation) adapter to the model (implies --no-mmap). This allows you to adapt the pretrained model to specific tasks or domains.
```

Should be fixed in #1145 too, don't reintroduce the same mistake here.

**Review by sw — APPROVED** [review]

**Pull request merged**